ktor-client/ktor-client-core/common/src/io/ktor/client/request/HttpRequest.kt

```diff
@@ -217,6 +217,39 @@ public class HttpResponseData constructor(
  */
 public fun HttpRequestBuilder.headers(block: HeadersBuilder.() -> Unit): HeadersBuilder = headers.apply(block)
 
+/**
+ * Executes a [block] that configures the [CookiesBuilder] associated to this request.
+ */
+public fun HttpRequestBuilder.cookies(block: CookiesBuilder.() -> Unit): HeadersBuilder =
+    CookiesBuilder(headers).apply(block).headersBuilder
+
+public class CookiesBuilder(internal val headersBuilder: HeadersBuilder) {
+    public fun append(
```

Comment:
The signature of this method is exactly duplicated by the class `Cookie`

Reply:
Yes that's correct, so what would are you suggesting?
Inside a `cookies { ... }` block we can also allow syntax like:
```
cookies {
     +Cookie(...constructor params...)
}
```
Or
```
cookies {
     append(Cookie(...constructor params...))
}
```
Does that sound better to you?

From my perspective, `+Cookie(...)` version is easy to write and handy, but not that discoverable.
While `append(Cookie(...))` version is discoverable but too verbose and visually redundant given that we are already inside a `cookies { ... }` block (and obviously what we are appending inside -- is cookies, so another word Cookie is redundant is not that good for readability).

Reply:
I don't like the syntax with `+Cookie(...)` cause it looks unfamiliar to Ktor users. 

I would keep `HttpRequestBuilder.cookie(...)` method